test/forest_test.cpp

```diff
@@ -78,11 +114,36 @@ auto big_test_forest() {
 
 /**************************************************************************************************/
 
+namespace detail {
+
+/**************************************************************************************************/
+
+template <typename T>
+inline auto to_string(const T& x) {
```

Comment:
I think inline is not needed here and in the following.

Reply:
Fixed